### DISCUSSION TITLE
Raffle.sol

### DISCUSSION THREAD

**Discussion by yasharyan0904** [Cyfrin/foundry-full-course-cu#1880] [Q&A] [answered]
CODE
```
function checkUpkeep(
        bytes memory 
    )
        public
        view
        override
        returns (bool upkeepNeeded, bytes memory )
    {
        bool isOpen = RaffleState.OPEN == s_raffleState;
        bool timePassed = ((block.timestamp - s_lastTimeStamp) > i_interval);
        bool hasPlayers = s_players.length > 0;
        bool hasBalance = address(this).balance > 0;
        upkeepNeeded = (timePassed && isOpen && hasBalance && hasPlayers);
        return (upkeepNeeded, "0x0"); // can we comment this out?
    }
    ```
    What is the bytes memory that we are taking as input for this function and its use?

**Comment by EngrPips** [comment, marked as answer]
Hello @yasharyan0904, If you set up an `upkeep` that needs some sort of external data to compute if upkeep is needed or not that is how you pass in the data that it required to do the computation. For our Lottery case, all the data needed to compute if the upkeep is needed or not is available in our contract state so we didn't use the `bytes memory` input. If anything is still confusing please ask questions.

**Reply by yasharyan0904** [reply]
So, in our function we are passing bytes memory but in this case (raffle) it has no data to pass!!
Right?

**Reply by EngrPips** [reply]
If i understand this statement correctly then my answer is `yes`

**Reply by yasharyan0904** [reply]
yep, THANKS A LOT:)

**Reply by EngrPips** [reply]
You are welcome, Friend. Keep crushing.
